### DISCUSSION TITLE
ldap-bind instead of comparing userPassword when authenticating

### DISCUSSION THREAD

**Discussion by svhan** [stalwartlabs/smtp-server#30] [Q&A]
Hello,

I'm very interested in migrating from postfix/dovecot to stalwart. My actual setup does not have the cleartext password stored in ldap, but a hash instead. It just does a bind as the authenticating user against the ldap and if this works, you're in.
Now, I'm wondering if this is possible with stalwart as well? I have no idea, how to set this up.

Thanks a lot for your help!

**Comment by mdecimus** [comment]
Hi,

LDAP bind authentication is not supported as it cannot handle SASL mechanisms such as SCRAM or OAUTH which require access to the password hash. There are however plans to add an option that disables SCRAM/OAUTH support in order to support LDAP bind auth.

**Comment by mdecimus** [comment]
This is now supported in `v0.5.0`.
